### PR TITLE
[breaking] promisify the api

### PR DESCRIPTION
**Pull request by SamVerschueren**
- Bump AVA and ES2015ify tests
- Add promise support and drop callback

### CODE REVIEW

**Comment by SamVerschueren** [comment]
Wondering why you use `gracefull-fs`. Any benefit here over the core `fs` module?

**Comment by sindresorhus** [comment]
> Wondering why you use gracefull-fs. 

Windows :(

File: index.js [hunk 1]
```diff
@@ -4,54 +4,46 @@ var fs = require('graceful-fs');
...
-			return;
-		}
-
-		var file = new File({
-			cwd: cwd,
-			base: base,
-			path: pth,
-			stat: stat
-		});
+	return pify(fs.stat.bind(fs), Promise)(pth)
```
Comment (sindresorhus) [inline index.js]: I don't think you need to `.bind()` here.

File: index.js [hunk 2]
```diff
@@ -4,54 +4,46 @@ var fs = require('graceful-fs');
...
+				file.contents = fs.createReadStream(pth).pipe(stripBomStream());
+				return [file];
+			}
 
-		if (opts.buffer === false) {
-			file.contents = fs.createReadStream(pth).pipe(stripBomStream());
-			cb(null, file);
-			return;
-		}
+			return Promise.all([file, pify(fs.readFile.bind(fs), Promise)(pth)]);
```
Comment (sindresorhus) [inline index.js]: This is pretty ugly. I think it would be better to just nest the promise.

Comment (sindresorhus) [inline index.js]: Like: https://gist.github.com/sindresorhus/df21f5b58e66309589cd/revisions

Comment (sindresorhus) [inline index.js]: Also would be better to just pify the whole `fs` module at the top.

Comment (SamVerschueren) [inline index.js]: That looks more cleaner indeed!

Didn't want to pify the entire module just for two methods.

Comment (sindresorhus) [inline index.js]: Not sure why you didn't want to, but the perf difference is absolutely neglible.

Comment (SamVerschueren) [inline index.js]: Because it iterates over all the methods. But you're right, easier to use and performance isn't an issue I guess.

**Comment by SamVerschueren** [comment]
Done! Much cleaner now, thanks :).

**Comment by sindresorhus** [comment]
Perfect! :)

**Pull request closed without merging**